core/shared/src/main/scala/zio/concurrent/ConcurrentSet.scala

```diff
@@ -11,15 +12,47 @@ final class ConcurrentSet[A] private (private val underlying: ConcurrentHashMap.
     UIO(underlying.add(x))
 
   def addAll(xs: Iterable[A]): UIO[Boolean] =
-    UIO {
-      var added = false
-      xs.foreach(x => added = underlying.add(x))
-      added
-    }
+    UIO(underlying.addAll(xs.asJavaCollection))
```

Comment:
Nice! And I didn't notice the subtle bug there (should have been `underlying.add(...) || added`. You may have add an annotation if compiler complains here.
```suggestion
    UIO(underlying.addAll(xs.asJavaCollection): @silent("JavaConverters"))
```